### PR TITLE
Development support php7.1

### PR DESCRIPTION
**Pull request by valdiearsanur**
status : `Ready To Review`
testing environment : checkout branch

* Update Guzzle to version 6.0 and OAuth Subscriber to 0.3
* KaskusClient refactor
* KaskusClient Unit Tests
* Class DIagram : [link](https://drive.google.com/a/gdplabs.id/file/d/1yJ0nTedTD2Eh9uAMEdX7QN4Yz2QL3By5/view?usp=sharing)
* Sequence DIagram (Basic Usage and Login) : [link](https://drive.google.com/a/gdplabs.id/file/d/1yJ0nTedTD2Eh9uAMEdX7QN4Yz2QL3By5/view?usp=sharing)

--------------------------
Testing Scenario : 
use this consumer key secret : 
$consumerKey = '[your valid cons key]';
$consumerSecret = 'your valid cons secret';

set this base url
$baseUri = '[BRANCHES_URL]';

1. run `sample/sample_php7.php`
- [ ] app give response

2. run `sample/simple_application`
- [ ] can login / logout using KASKUS
- [ ] can retrieve KASKUS profile

2. run unit test 
- [ ] passed unit test

### CODE REVIEW

File: sample/sample.php [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
-$consumerKey = 'YOUR_API_KEY';
-$consumerSecret = 'YOUR_API_SECRET';
 
-$client = new \Kaskus\KaskusClient($consumerKey, $consumerSecret);
+//todo: revert this
+// $consumerKey = 'YOUR_API_KEY';
+// $consumerSecret = 'YOUR_API_SECRET';
+
+$consumerKey = '03f0968bdbd3462e77ff719b717f40';
+$consumerSecret = 'b0b1338bcda983fe77342bab138951';
```
Comment (zulamdat) [inline sample/sample.php]: @valdiearsanur revert val biar gak kelupaan

File: sample/simple_application/index.php [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+session_start();
+
+// configuration
+//todo: revert this
+//$consumerKey = 'YOUR_API_KEY';
+//$consumerSecret = 'YOUR_API_SECRET';
+//$callbackUrl = 'http://localhost:8000'; // e.g. http://yourapplication.com
+$consumerKey = '03f0968bdbd3462e77ff719b717f40';
+$consumerSecret = 'b0b1338bcda983fe77342bab138951';
+$callbackUrl = 'http://valdie.phpsdk-kaskus.dev/index.php'; // e.g. http://yourapplication.com
```
Comment (zulamdat) [inline sample/simple_application/index.php]: @valdiearsanur ini jg val, revert aja, ntar pas ngetestnya biar edit sendiri

File: sample/simple_application/index.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+<html>
+<head>
+	<title>KASKUS PHP SDK Example</title>
+</head>
+<body>
+	<?php if ($authorized): ?>
+		<div>
+			<p>Successfully login with KASKUS, here's your token: </p>
+			<p>Token : <?= $token_key; ?></p>
+			<p>Token Secret : <?= $token_secret; ?></p>
```
Comment (zulamdat) [inline sample/simple_application/index.php]: @valdiearsanur <?php echo aja val

File: sample/simple_application/kaskus_profile.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<?php
+require __DIR__ . '/vendor/autoload.php';
+session_start();
+
+$consumer_key = '03f0968bdbd3462e77ff719b717f40';
+$consumer_secret = 'b0b1338bcda983fe77342bab138951';
```
Comment (zulamdat) [inline sample/simple_application/kaskus_profile.php]: @valdiearsanur ini jg val

File: src/Kaskus/BaseKaskusClient.php [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+class BaseKaskusClient
+{
+	const UNAUTHENTUCATED_STACK = 'unauthenticated';
+	const AUTHENTUCATED_STACK = 'authenticated';
+
+	use HasHandlerStackTrait;
+
+	//todo: revert this
+	//protected $baseUri = 'https://www.kaskus.co.id/api/oauth/';
```
Comment (zulamdat) [inline src/Kaskus/BaseKaskusClient.php]: @valdiearsanur 💡 reminder

File: src/Kaskus/KaskusClient.php [hunk 6]
```diff
@@ -1,126 +1,30 @@
...
+		$baseUri = null
+	) {
+		$this->consumerKey = $consumerKey;
+		$this->consumerSecret = $consumerSecret;
+
+		if (isset($baseUri)) {
+			$this->baseUri = $baseUri;
+		}
+
+		$cientFactory = new ClientFactory();
```
Comment (zulamdat) [inline src/Kaskus/KaskusClient.php]: @valdiearsanur typo val `$clientFactory`

File: src/Kaskus/OAuthFactory.php [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?php
+namespace Kaskus\Client;
+
+use GuzzleHttp\Subscriber\Oauth\Oauth1;
+
+class OAuthFactory
```
Comment (zulamdat) [inline src/Kaskus/OAuthFactory.php]: @valdiearsanur better OAuth1Factory gak ya namanya? biar kalau nanti ada yang oauth2 gak bentrok, gimana enaknya?

File: src/Kaskus/BaseKaskusClient.php [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+use Kaskus\Exceptions\UnauthorizedException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class BaseKaskusClient
+{
+	use HasHandlerStackTrait;
+
+	const UNAUTHENTUCATED_STACK = 'unauthenticated';
+	const AUTHENTUCATED_STACK = 'authenticated';
```
Comment (zulamdat) [inline src/Kaskus/BaseKaskusClient.php]: @valdiearsanur unauthentucated dan authentucated 🤣

**Comment by valdiearsanur** [comment]
minta tolong review ya ka @kawakibi @anggara-kaskus :D

**Pull request closed without merging**